File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int begin=0;
        int sum=sequence[0];
        if(sum==k) return answer;
        answer[1]=sequence.length-1;
        for(int i=1;i<sequence.length;i++){
            sum+=sequence[i];
            while(sum>=k){
                if(sum==k&&i-begin<answer[1]-answer[0]){
                    answer[0]=begin;
                    answer[1]=i;
                }
                sum-=sequence[begin++];
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let seqStart=0;
    let seqEnd=k;
    let currentStart=0;
    let currentSum=0;
    for(let i=0;i<sequence.length;i++){
        currentSum+=sequence[i];
        while(currentSum>k){
            currentSum-=sequence[currentStart++];
        }
        if(currentSum===k&&i-currentStart<seqEnd-seqStart){
            seqStart=currentStart;
            seqEnd=i;
        }
    }
    return [seqStart, seqEnd];
}